### ISSUE TITLE
Add EQL gridder in geodetic coordinates

### ISSUE THREAD

**Issue by santisoler** [fatiando/harmonica#175]
**Description of the desired feature**

Would be nice to have an equivalent layer gridder that works in geodetic coordinates.
This means that the data points and the location of the point sources should be defined in geodetic coordinates.
The Green's function should still be equal to 1 over the Euclidean distance between each data point and each source.
After #154 is closed, we will be able to easily compute this distance between two given points defined in geodetic coordinates.
The new EQL gridder should also take the reference ellipsoid, which should be a instance of `boule.Ellipsoid`.
Would be better to wait until #165 is fixed before start working on this issue: the idea is to have a common base class for every EQL gridder.

After having a geodetic EQL, the spherical EQL could be ditched.
There are very few use cases for an EQL spherical: data is generally given in geodetic coordinates and grids are usually expected to be defined in the same coordinate system. Besides, having a single EQL class that handles gridding while taking into account the curvature of the Earth it's clearer for users.

**Are you willing to help implement and maintain this feature?** Yes
